youtube_dl/extractor/bandlab.py

```diff
@@ -0,0 +1,191 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+
+
+class BandlabIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?bandlab\.com/post/(?P<id>[^/]+)'
+
+    _TEST = {
+        'url': 'https://www.bandlab.com/post/f5f04998635a44ea819cacdba7ae2076_f8d8574c3bdaec11b6562818783151a1',
+        'info_dict': {
+            'id': 'f5f04998635a44ea819cacdba7ae2076_f8d8574c3bdaec11b6562818783151a1',
+            'ext': 'm4a',
+            'title': 'ON MY OWN (unreleased)',
+            'artist': 'Michael MacDonald',
+        },
+    }
+
+    def _real_extract(self, url):
+        track_id = self._match_id(url)
+        config = self._download_json(
+            'http://www.bandlab.com/api/v1.3/posts/%s' % track_id, track_id)
+        track_url = config['track']['sample']['audioUrl']
```

Comment:
Condition the value:
```suggestion
        track_url = url_or_none(config['track']['sample']['audioUrl'])
```

Reply:
Actually this is important. If the site sends something that isn't a URL, the extractor should raise an exception; otherwise the user gets a confusing error from the core processing.